src/main/java/com/google/api/codegen/transformer/csharp/CSharpGapicClientPackageTransformer.java

```diff
@@ -0,0 +1,140 @@
+  private final FileHeaderTransformer fileHeaderTransformer =
+      new FileHeaderTransformer(new StandardImportSectionTransformer());
+  private final PackageMetadataTransformer metadataTransformer = new PackageMetadataTransformer();
+
+  public CSharpGapicClientPackageTransformer(
+      GapicCodePathMapper pathMapper, PackageMetadataConfig packageMetadataConfig) {
+    this.pathMapper = pathMapper;
+    this.packageMetadataConfig = packageMetadataConfig;
+  }
+
+  @Override
+  public List<ViewModel> transform(ProtoApiModel model, GapicProductConfig productConfig) {
+    List<ViewModel> surfaceDocs = new ArrayList<>();
+    SurfaceNamer namer = new CSharpSurfaceNamer(productConfig.getPackageName(), ALIAS_MODE);
+    CSharpFeatureConfig featureConfig = new CSharpFeatureConfig();
+
+    InterfaceModel lastApiInterface = null;
+    for (InterfaceModel apiInterface : model.getInterfaces()) {
+      lastApiInterface = apiInterface;
+    }
+
+    GapicInterfaceContext context =
+        GapicInterfaceContext.create(
+            lastApiInterface,
+            productConfig,
+            csharpCommonTransformer.createTypeTable(productConfig.getPackageName(), ALIAS_MODE),
+            namer,
+            featureConfig);
+    surfaceDocs.add(generateCsProjView(context));
+
+    return surfaceDocs;
+  }
+
+  @Override
+  public List<String> getTemplateFileNames() {
+    return Arrays.asList(CSPROJ_TEMPLATE_FILENAME);
+  }
+
+  private PackageInfoView generateCsProjView(GapicInterfaceContext context) {
+    Model model = context.getModel();
+    GapicProductConfig productConfig = context.getProductConfig();
+    PackageInfoView.Builder view = PackageInfoView.newBuilder();
+    view.templateFileName(CSPROJ_TEMPLATE_FILENAME);
+    String outputPath =
+        pathMapper.getOutputPath(context.getInterface().getFullName(), productConfig);
+    view.outputPath(outputPath + File.separator + productConfig.getPackageName() + ".csproj");
+    view.fileHeader(fileHeaderTransformer.generateFileHeader(context));
+    view.serviceTitle(model.getServiceConfig().getTitle());
+    view.serviceDescription(model.getServiceConfig().getDocumentation().getSummary().trim());
+    view.domainLayerLocation(productConfig.getDomainLayerLocation());
+    view.authScopes(new ArrayList<>()); // Unused in C#
+    view.releaseLevel(productConfig.getReleaseLevel());
+    String versionSuffix;
+    switch (productConfig.getReleaseLevel()) {
+      case ALPHA:
+        versionSuffix = "-alpha01";
+        break;
+      case BETA:
+        versionSuffix = "-beta01";
+        break;
+      default:
+        versionSuffix = "";
+        break;
+    }
+    view.version("1.0.0" + versionSuffix);
+    String tags = "";
+    for (String tag : Splitter.on('.').split(productConfig.getPackageName())) {
+      if (tag.matches("[vV][\\d.]+")) {
+        break;
+      }
+      tags += ";" + tag;
+    }
+    view.tags(tags.isEmpty() ? "" : tags.substring(1));
```

Comment:
This process is a bit opaque on first read. If you filter the matching tags out of the split package name, you can just use a `Joiner` to intersperse the semicolons.